File: examples/05_parameterized/parameterized.py
"""Example 05: Parameterized Circuit — Parameters and expressions."""

from qiskit import QuantumCircuit
from qiskit.circuit import Parameter

qc = QuantumCircuit(2, 2)

# Create symbolic parameters
theta = Parameter('theta')
phi = Parameter('phi')

# Use parameters directly in gates
qc.h(0)
qc.rx(theta, 0)
qc.ry(phi, 1)
qc.cx(0, 1)
qc.measure([0, 1], [0, 1])

print("Parameterized circuit:")
print(qc)
print(f"\nParameters: {qc.parameters}")

# Parameter expressions support arithmetic
a = theta + 0.5
b = phi * 2.0
print(f"\ntheta + 0.5 = {a}")
print(f"phi * 2.0   = {b}")

File: examples/05_parameterized/parameterized.cpp
// Example 05: Parameterized Circuit — Parameters and binding.

#include "circuit/quantumcircuit.hpp"
#include <cmath>

using namespace Qiskit::circuit;

int main() {
    QuantumRegister qr(2);
    ClassicalRegister cr(2);
    QuantumCircuit circ(qr, cr);

    Parameter theta("theta");
    Parameter phi("phi");

    circ.h(0);
    circ.rx(theta, 0);
    circ.ry(phi, 1);
    circ.cx(0, 1);
    circ.measure(qr, cr);

    std::cout << "Parameterized circuit:" << std::endl;
    circ.print();

    // Bind parameters
    circ.assign_parameters({"theta", "phi"}, {M_PI / 4, M_PI / 3});

    std::cout << "\nBound circuit:" << std::endl;
    circ.print();

    return 0;
}
